dev/mongosh.ts

```diff
@@ -0,0 +1,230 @@
+// https://docs.mongodb.com/mongodb-shell/reference/options/
+
+import { decodedTextSpanIntersectsWith } from "typescript";
+
+const completionSpec: Fig.Spec = {
+  name: "mongosh",
+  description:
+    "The MongoDB Shell, mongosh , is a fully functional JavaScript and Node.js 14.x REPL environment for interacting with MongoDB deployments",
+  args: {
+    name: "Connection String",
+    isOptional: true,
+    default: "mongodb://localhost:27017",
+    suggestions: [
+      {
+        name: "mongodb://localhost:27017",
+        description:
+          "Default Connection String; Equivalant to running mongosh without any commands",
+      },
+      {
+        name: "mongodb+srv://cluster0.example.mongodb.net/sample_geospatial",
+        description: "Atlas Connection String Example",
+        priority: 35,
+      },
+    ],
+  },
+
+  options: [
+    {
+      name: ["-v", "--version"],
+      description: "View your current mongosh version",
+    },
+    {
+      name: "--shell",
+      description:
+        "Enables the shell interface. If you invoke the mongosh command and specify a JavaScript file as an argument, or use --eval to specify JavaScript on the command line, the --shell option provides the user with a shell prompt after the file finishes executing",
+    },
+    {
+      name: "--eval",
+      description:
+        "<javascript>: Evaluates a JavaScript expression that is specified as an argument. mongosh does not load its own environment when evaluating code. As a result many options of the shell environment are not available. The result of evaluating a JavaScript argument is printed to your command line",
```

Comment:
I don't think the `<javascript>: ` part is needed since the description explains well how it works:
```suggestion
        "Evaluates a JavaScript expression that is specified as an argument. mongosh does not load its own environment when evaluating code. As a result many options of the shell environment are not available. The result of evaluating a JavaScript argument is printed to your command line",
```